bin/wasm-node/rust/src/network_service.rs

```diff
@@ -235,6 +236,21 @@ impl NetworkService {
         result
     }
 
+    /// Announces transaction to the peers we are connected to.
+    // TODO: more docs
+    pub async fn announce_transaction(
+        self: Arc<Self>,
+        transaction: &[u8],
+    ) -> Result<(), QueueNotificationError> {
+        for target in self.peers_list().await {
+            let _ = self
+                .network
+                .announce_transaction(&target, 0, &transaction)
+                .await;
+        }
+        Ok(())
+    }
```

Comment:
```suggestion
    /// Announces transaction to the peers we are connected to.
    ///
    /// Returns an error if we aren't connected to any peer, or if we fail to send the transaction to all peers.
    pub async fn announce_transaction(
        self: Arc<Self>,
        transaction: &[u8],
    ) -> Result<(), ()> {  // TODO: proper error type
        let mut any_propagated = false;
    
        for target in self.peers_list().await {
            if self
                .network
                .announce_transaction(&target, 0, &transaction)
                .await.is_ok() { any_propagated = true; }
        }

        if any_propagated { Ok(()) } else { Err(()) }
    }
```

(I didn't format the code)